### PR TITLE
Colin/general cleanup

### LINKED ISSUE #585: feat: improve error handling for forking
# Idea
With all the functionality added with the forking feature, it was done so with mostly unwraps. We should add some better error handling/logging. 

## Solution
Structure errors as we have been with ThisError and take best use of the #[from] macro.

### PR DESCRIPTION
**Pull request by Autoparallel**
**Give an overview of the tasks completed**
Just cleaning up throughout the codebase. Made some changes that I am not married to. Please review!

Closes #585

### CODE REVIEW

**Review by 0xJepsen — APPROVED** [review]
This looks clean to me, I don't have a problem merging this in

**Comment by 0xJepsen** [comment]
Can this close #585?

**Comment by Autoparallel** [comment]
Yes

File: bin/init.rs [hunk 1]
```diff
@@ -111,12 +113,7 @@ pub(crate) fn init_project(name: &str) -> io::Result<()> {
...
 ///   issues.
-pub fn remove_git() -> Result<(), Box<dyn Error>> {
-    let status = Command::new("rm").arg("-rf").arg(".git").status()?;
-
-    if !status.success() {
-        return Err("failed to execute process".into());
-    }
-
+pub fn remove_git() -> Result<(), ArbiterError> {
+    Command::new("rm").arg("-rf").arg(".git").status()?;
```
Comment (kinrezC) [inline bin/init.rs:117]: I know this isn't part of this PR but rm -rf .git as a public function is a bit scary, maybe there's a safer way to do this?

**Review by kinrezC — APPROVED** [review]
looks good

**Review by Alexangelj — APPROVED** [review]
LGTM - always good to see unwraps removed

**Pull request merged**